packages/wrapped-keys/src/lib/litActions/solana/src/signMessageWithSolanaEncryptedKey.js

```diff
@@ -0,0 +1,70 @@
+const { Keypair } = require('@solana/web3.js');
+const bs58 = require('bs58');
+const nacl = require('tweetnacl');
+
+(async () => {
+  const LIT_PREFIX = 'lit_';
+
+  let decryptedPrivateKey;
+  try {
+    decryptedPrivateKey = await Lit.Actions.decryptToSingleNode({
+      accessControlConditions,
+      chain: 'ethereum',
+      ciphertext,
+      dataToEncryptHash,
+      authSig: null,
+    });
+  } catch (error) {
+    Lit.Actions.setResponse({
+      response: `Error decrypting data to private key: ${error.message}`,
+    });
+    return;
+  }
+
+  if (!decryptedPrivateKey) {
+    // Exit the nodes which don't have the decryptedData
+    return;
+  }
+
+  const privateKey = decryptedPrivateKey.startsWith(LIT_PREFIX)
+    ? decryptedPrivateKey.slice(LIT_PREFIX.length)
+    : decryptedPrivateKey;
+  const solanaKeyPair = Keypair.fromSecretKey(
+    Uint8Array.from(Buffer.from(privateKey, 'hex'))
```

Comment:
Why do we need a buffer? We don't need it if we just start by encrypting it as a hex?

Reply:
Updated to
```ts
const solanaKeyPair = Keypair.fromSecretKey(Buffer.from(privateKey, 'hex'));
```
It is encrypted as hex, but the Solana library requires it to be passed as a `Uint8Array`